client/jvm/src/main/scala/org/http4s/client/JavaNetClientBuilder.scala

```diff
@@ -139,6 +140,14 @@ sealed abstract class JavaNetClientBuilder[F[_]] private (
   def resource: Resource[F, Client[F]] =
     Resource.eval(F.delay(create))
 
+  private def normalizeRequest(req: Request[F]): Request[F] =
+    if (req.method === Method.GET) {
```

Comment:
I left HEAD out of scope for this PR - happy to look into this afterwards though.